client/fingerprint/cpu_linux.go

```diff
@@ -0,0 +1,9 @@
+package fingerprint
+
+import (
+	"github.com/hashicorp/nomad/client/lib/cgutil"
+)
+
+func (f *CPUFingerprint) deriveReservableCores(req *FingerprintRequest) ([]uint16, error) {
+	return cgutil.GetCPUsFromCgroup(req.Config.CgroupParent)
```

Comment:
Looks like we don't really test the client/lib utils, but it may be nice to have a linux specific `cpu_linux_test`

Reply:
Addressed in #10328